test/WebJobs.Extensions.DurableTask.Analyzers.Test/EntityInterface/EntityInterfaceReturnTypeAnalyzerTests.cs

```diff
@@ -128,21 +130,104 @@ public interface IEntityExample
             VerifyCSharpDiagnostic(test, expectedDiagnostics);
         }
 
+        // Tests SyntaxKind.GenericName
+        [TestMethod]
+        public void ReturnTypeAnalyzer_IncorrectReturn_List()
+        {
+            var test = @"
+    using System.Threading.Tasks;
+    using System.Collections.Generic;
+    using Microsoft.Azure.WebJobs;
+    using Microsoft.Azure.WebJobs.Extensions.DurableTask;
+
+    namespace VSSample
+    {
+        public static class HelloSequence
+        {
+            [FunctionName(""ReturnTypeAnalyzerTestCases"")]
+            public static async Task Run(
+            [OrchestrationTrigger] IDurableOrchestrationContext context)
+            {
+                context.SignalEntityAsync<IEntityExample>();
```

Comment:
I'm a bit confused about what this diagnostic is trying to detect. First, it looks like `SignalEntityAsync()` is [missing some required parameters](https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.webjobs.extensions.durabletask.durableentityproxyextensions.signalentityasync?view=azure-dotnet#Microsoft_Azure_WebJobs_Extensions_DurableTask_DurableEntityProxyExtensions_SignalEntityAsync__1_Microsoft_Azure_WebJobs_Extensions_DurableTask_IDurableEntityClient_Microsoft_Azure_WebJobs_Extensions_DurableTask_EntityId_System_Action___0__). Second, is the concern that the customer is calling an entity method with the signal functionality that has a return value? I don't know if that is necessarily an anti-pattern.

Reply:
These EntityInterface tests are actually analyzing the entity interfaces and producing diagnostics on the interface code (aside from InterfaceAnalyzer itself, which is the initial analyzer that runs to store the entity interfaces and trigger the other analyzers to run afterward. This analyzer can also produce diagnostics on this code if the type argument on SignalEntityAsync is not an interface that is defined in user code). What I mean is that these are primarily analyzers for the entity interface itself:

- content (only methods)

- return type (methods must only have the return types void, Task, or Task<T>)

- parameters (methods must have one or 0 parameters).

`public interface IEntityExample
        {
            public static void methodTestVoid();

            public static Task methodTestTask(string test);

            public static Task<T> methodTestTaskT(string test);
        }`

My analyzers determine that an interface is an "entity interface" by saving the user defined interface that is used in SignalEntityAsync. I know that SignalEntityAsync does use parameters, but these tests are for the type argument on SiganlEntityAsync specifically. Also, the test is only comparing the diagnostics before and after my analyzer runs (also, to be clear, it checks that no other analyzer that I wrote produced a diagnostic that I did not intend in the test). I thought it would be extra clutter to add the arguments and would take away from the understanding of the test. I can add the parameters, but I didn't think it was necessary.

I can also write comments on the top of these tests to explain what they are testing if that would help.

Reply:
I'm actually not sure if adding comments to my tests would be better for easier understanding or would be worse because the comments would potentially be out of date when logic in the analyzers changes. Do you have an opinion on this?